src/services/destination/nativeIntegration.ts

```diff
@@ -106,7 +106,10 @@ export default class NativeIntegrationDestinationService implements IntegrationD
         );
         try {
           const doRouterTransformationResponse: RouterTransformationResponse[] =
-            await destHandler.processRouterDest(cloneDeep(destInputArray));
+            await destHandler.processRouterDest(
+              cloneDeep(destInputArray),
+              cloneDeep(requestMetadata),
```

Comment:
Do we need to do clone deep we are not mutating this and reusing this

Reply:
addressed